quickwit-indexing/src/merge_policy.rs

```diff
@@ -146,27 +146,27 @@ pub trait MergePolicy: Send + Sync + fmt::Debug {
 #[derive(Clone, Debug)]
 pub struct StableMultitenantWithTimestampMergePolicy {
     /// We never merge segments larger than this size.
-    pub max_merge_docs: usize,
-    pub min_level_num_docs: usize,
-    pub merge_factor: usize,
-    pub merge_factor_max: usize,
+    pub demux_enabled: bool,
     pub demux_factor: usize,
     pub demux_field_name: Option<String>,
+    pub min_level_num_docs: usize,
     pub merge_enabled: bool,
-    pub demux_enabled: bool,
+    pub merge_factor: usize,
+    pub max_merge_factor: usize,
+    pub split_max_num_docs: usize,
```

Comment:
It was not named `split_max_num_docs` because we can exceed this size actually.
We do not consider splits over this size for merge however.

Anyway I am ok with this name.

Reply:
```suggestion
    /// This is not really a "max".
    ///
    /// The merge policy aims at eventually producing mature splits that 
    /// have size larger but close to `split_max_num_docs`.
    /// We never merge segments larger or equal to `split_max_num_docs`.
    pub split_max_num_docs: usize,
```